cogs/boolean_utils.py

```diff
@@ -0,0 +1,395 @@
+        process the expression by passing in the dictionary which is
+        keeping track of the value of each variable
+        '''
+        if depth >= len(self.var_set):
+            # Process expression for a single permutation of input
+            self.process_single_exp(var_dict, vars)
+        else:
+            # Continue recursion
+            var_dict[vars[depth]] = False
+            self.recurse_through_var(vars, var_dict, depth + 1)
+            var_dict[vars[depth]] = True
+            self.recurse_through_var(vars, var_dict, depth + 1)
+
+    def process_single_exp(self, dict_bool, vars):
+        '''
+        Processes a post-fix expression with the values of each variable
+        stroed in `dict_bool`.
+        '''
+        post_stack = []
+        for char in self.post_exp:
+            if char.isalpha():
+                # Append value of that variable to the stack
+                post_stack.append(dict_bool[char])
+            elif char in self.prec_dict:
+                # Compute result and add it back to stack
+                if char == '!':
+                    one = not post_stack.pop()
+                    post_stack.append(one)
+                if char == '*':
+                    one = post_stack.pop()
+                    two = post_stack.pop()
+                    result = one and two
+                    post_stack.append(result)
+                if char == '+':
+                    one = post_stack.pop()
+                    two = post_stack.pop()
+                    result = one or two
+                    post_stack.append(result)
+                if char == '^':
+                    one = post_stack.pop()
+                    two = post_stack.pop()
+                    result = one and not two or two and not one
+                    post_stack.append(result)
+        # Save result in a formatted fashion, will be single line in table
+        self.format_result(post_stack.pop(), dict_bool, vars)
+
+    def format_header(self, vars):
+        '''Creates the header to the truth table'''
+        self.result_formatted += "+" + "---+" * len(vars)
+        self.result_formatted += "-" * len(self.disp_exp) + "--+\n"
+        for var in vars:
+            self.result_formatted += "| {} ".format(var)
+        self.result_formatted += "| {} |\n".format(self.disp_exp)
+        self.result_formatted += "+---" * len(vars)
+        self.result_formatted += "+" + "-" * len(self.disp_exp) + "--+\n"
+
+    def format_result(self, result, dict_bool, vars):
+        '''Creates single line in the truth table'''
+        for var in vars:
+            booly = self.bool_to_int(dict_bool[var])
+            self.result_formatted += "| {} ".format(booly)
+        res = self.bool_to_int(result)
+        self.result_formatted += "| {}".format(res)
+        self.result_formatted += " " * len(self.disp_exp) + "|\n"
+
+    def bool_to_int(self, booly):
+        '''Returns 1 if True, 0 if False'''
```

Comment:
```py
>>> int(True)
1
>>> int(False)
0
```

Let's use int() instead.

Reply:
Gotchya, I didn't know that worked like that